optional-requirements.txt

```diff
@@ -1,2 +1,3 @@
 numpy>=1.10.2,<=2.0.0
 pandas>=0.17.1,<=3.0.0
+PyArrow>=11.0.0,<16.0.0
```

Comment:
FYI, i think enough time has passed that we can drop 3.7 support. The intent of tox is to support the most recent 3 python versions plus the most recent pypy version. Maybe a sep PR bumping up the support py versions.

Reply:
SGTM. What's the right thing to put as the lower bound? If it were me, I'd just put PyArrow without any version specification.

Reply:
I went with no lower bound for now. We can add one if needed.

Reply:
It's awkward. Currently, PyArrow is an optional dependency, but without it, pandas raises a DeprecationWarning (https://github.com/pandas-dev/pandas/issues/54466). Once pandas 3 is out and can be relied upon, it will include the dependency and can be dropped from this project.